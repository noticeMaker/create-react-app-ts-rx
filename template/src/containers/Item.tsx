import * as React from 'react';

import { IProductState } from 'src/reducers/product'

interface IProps extends IProductState {
    propName?: any
}

class Item extends React.Component<IProps> {

    public render() {
        return (
            <div className="item">
                {this.props.product.map((v, idx) => {
                   return <li key={idx} className={`${idx}o`}>{v.id}</li>
               })}
            </div>
        );
    }
}

export default Item