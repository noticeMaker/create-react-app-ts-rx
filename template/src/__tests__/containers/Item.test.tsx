
import * as React from "react";
import { shallow } from "enzyme";

import Item from "src/containers/Item"

const source = [
    { "id": 1, "title": "iPad 4 Mini", "price": 500.01, "inventory": 2 },
]

test("<Item />", () => {

    const wrapper = shallow(<Item product={source}/>);
    expect(wrapper).toMatchSnapshot();
});