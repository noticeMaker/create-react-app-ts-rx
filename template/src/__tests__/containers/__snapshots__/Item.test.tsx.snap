// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    product={
        Array [
            Object {
              "id": 1,
              "inventory": 2,
              "price": 500.01,
              "title": "iPad 4 Mini",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="0o"
>
          1
</li>,
      ],
      "className": "item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "0o",
        },
        "ref": null,
        "rendered": 1,
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="0o"
>
            1
</li>,
        ],
        "className": "item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "0o",
          },
          "ref": null,
          "rendered": 1,
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
